nibabel/nicom/utils.py

```diff
@@ -27,26 +27,20 @@ def find_private_section(dcm_data, group_no, creator):
     Returns
     -------
     element_start : int
-        Element number at which named section starts
+        Element number at which named section starts.
     """
-    is_regex = hasattr(creator, 'search')
-    if not is_regex:  # assume string / bytes
+    if hasattr(creator, 'search'):
+        match_func = lambda x : creator.search(x)
```

Comment:
Unneeded indirection.

```suggestion
        match_func = creator.search
```